devices.php

```diff
@@ -910,6 +910,9 @@ function buildCustomAttributes($template, $device) {
 					print "\n\t<option value=\"$dcaValue\"$selected>$dcaValue</option>";
 				}
 				echo '</select></div>';
+			} else if ($cvtype=="url") {
+				echo '<div><input type="text"',$validation,' name="',$inputname,'" id="',$inputname,'" value="',$customdata["value"],'">';
+				echo '<button type="button" onclick=window.location.href="',$customdata["value"],'" value="open">Open</button></div>';

```

Comment:
Come to think of it the way we handle the buttons that auto hide on the device templates page would be an excellent way to handle this.  The button would only appear when there is valid content for it to work off of.